core/include/thread_flags.h

```diff
@@ -61,13 +65,40 @@
  * @name reserved thread flags
  * @{
  */
+
+#if defined(MODULE_CORE_THREAD_FLAGS) || defined(DOXYGEN)
+/**
+ * @brief Set when a message becomes available
+ *
+ * This flag is set for a thread if a message becomes available either by being
+ * queued in the thread's message queue or by another thread becoming
+ * send-blocked.
+ *
+ * It is only reset through thread_flags_wait_*(), not by msg_receive().
+ * Care must be taken to a) not lose message notifications and b) not block for
+ * messages but not thread flags.
```

Comment:
difficult to understand what you mean by "not block messages but not thread flags."

Reply:
does "not block the thread for messages without blocking for flags" describe what you meant?
(`bl rx` vs. `bl anyfl`)

Reply:
I've completely reworded the not, is it clearer like this?